views/js/qtiCreator/helper/saveChanges.js

```diff
@@ -27,6 +29,9 @@ define(function () {
      * @returns {Promise}
      */
     return itemCreator => new Promise((resolve, reject) => {
+        //trigger event to restore all currently active widget back to sleep state
+        $('#item-editor-panel').trigger('beforesave.qti-creator');
```

Comment:
I tried to debug and don't see when this event triggered. For save we have already this event and it is here https://github.com/oat-sa/extension-tao-itemqti/blob/4cecb90ad083160d2b7adf9ae01cd0cdf2a088e6/views/js/qtiCreator/plugins/menu/save.js#L58
Can you give me steps or describe logic when we need this event here?

Reply:
1. Make changes in item
2. Click on `Manage Items`
3. Click on `Save` in opened dialog